src/Eccube/Controller/README.html

```
@@ -0,0 +1,92 @@
+  <li>ルーティングは <strong><code>#[Route]</code> 属性</strong>（<code>Symfony\Component\Routing\Attribute\Route</code>）。管理画面のパスは <code>%eccube_admin_route%</code> プレフィックス配下に置く。</li>
+  <li>リクエスト／レスポンスのライフサイクルへの割り込みは <strong>EventSubscriber / EventListener</strong> で行う（コントローラを直接いじらない）。</li>
+ </ul>
+ <p>実装の書き方・DO/DON'T は <a href="../../../.claude/skills/controller/SKILL.md">controller/SKILL.md</a> が正典です。</p>
+</section>
+
+<section data-section="notes" data-customer="false">
+ <h2>注意点<span class="dev-only">開発者向け</span></h2>
+ <ul>
+  <li><strong>Fat コントローラを増やさない。</strong> 金額・在庫・送料・ポイント等の計算、<code>$em->persist()</code>/<code>$em->flush()</code> を伴う業務的な永続化、複数 Repository をまたぐ処理・外部 API・メール送信・ファイル IO がアクションに混ざり始めたら Service（受注処理は PurchaseFlow）へ抽出する。行数で線引きしない。</li>
+  <li><strong>認可はファイアウォール＋ロールで制御。</strong> 管理アクションは必ず <code>%eccube_admin_route%</code> 配下に置く（admin ファイアウォールで認証が要求される）。EC-CUBE コアは個別の <code>#[IsGranted]</code> ではなくファイアウォールで守る点に注意。</li>
+  <li><strong>CSRF: GET 以外の状態変更はトークン検証。</strong> フォーム経由（<code>handleRequest()</code>＋<code>isValid()</code>）は保護込み。フォームを介さない削除・Ajax は基底の <code>$this->isTokenValid()</code> を明示的に呼ぶ（無効時は <code>AccessDeniedHttpException</code>）。<code>isTokenValid()</code> はトークンをリクエストパラメータ <code>Constant::TOKEN_NAME</code> だけでなく <code>ECCUBE-CSRF-TOKEN</code> ヘッダからも受け付けるため、Ajax はヘッダ送信でよい。</li>
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ > ].

(spec-char-escape)